chainer/functions/loss/negative_sampling.py

```diff
@@ -104,20 +114,44 @@ def forward_gpu(self, inputs):
             ''',
             'negative_sampling_forward'
         )(self.wx, n_in, self.sample_size + 1, self.ignore_mask[:, None])
+
         if self.reduce == 'sum':
             loss = loss.sum()
         else:  # 'no':
             loss = loss.sum(axis=1)
         return loss,
 
-    def backward_cpu(self, inputs, grads):
-        x, t, W = inputs
-        gloss, = grads
+    def backward(self, indexes, grad_outputs):
+        x, t, W = self.get_retained_inputs()
+        gy, = grad_outputs
+        grad = NegativeSamplingFunctionGrad(
+            self.reduce, self.ignore_mask, self.sample_size, self.samples,
+            self.wx)
+        grad.ignore_label = self.ignore_label
+        return grad.apply((x, W, gy))
+
+
+class NegativeSamplingFunctionGrad(function_node.FunctionNode):
+
+    ignore_label = -1
```

Comment:
`ignore_label` looks not used in the grad function.

Reply:
Thanks, I removed the variable and the assignment in the main class.